src/components/LoginScreen.js

```diff
@@ -1,28 +1,122 @@
 import React from 'react';
-import {Grid, Button, Form} from 'semantic-ui-react';
+import LoadingComponent from './Loading';
+import { Link, useHistory } from 'react-router-dom';
+import { useAuthContext } from '../contexts/Firebase/AuthContext';
+import {Grid, Message, Button, Form} from 'semantic-ui-react';
 
 const LoginScreen = () => {
+  const history = useHistory();
+  const { auth, googleProvider } = useAuthContext();
+
+  const [ loading, setLoading ] = React.useState(false);
+
+  const loginWithGoogle = () => {
+    setLoginError(null);
+    setLoading(true);
+
+    auth
+      .signInWithPopup(googleProvider)
+      .then(_ => {
+        history.push('/');
+      })
+      .finally(_ => setLoading(false))
+  }
+
+  const loginWithEmailPassword = (e) => {
+    e.preventDefault();
+
+    setLoading(true);
+    setLoginError(null);
+    auth
+      .signInWithEmailAndPassword(email, password)
+      .then(_ => {
+        history.push('/');
+      })
+      .catch(e => {
+        if ( e.code === "auth/user-not-found" ||
+            e.code === "auth/wrong-password") {
+          setLoginError('Wrong Email/Password entered')
+        } else {
+          setLoginError(e.message);
+        }
+      })
+      .finally(_ => setLoading(false))
+  }
+
+  const [loginError, setLoginError] = React.useState(null);
+  let loginMessage;
+  if (loginError) {
+    loginMessage =
+      <Grid.Row>
+        <Grid.Column>
+          <Message negative>{loginError}</Message>
+        </Grid.Column>
+      </Grid.Row>
+  }
+
+  const [emailTouched, setEmailTouched] = React.useState(false);
```

Comment:
What are the touched variables for?

Reply:
To prevent validation from showing up until the user has begun typing in the field.

Reply:
ah ok